tests/integration/security/multi_signers/main_test.go

```diff
@@ -0,0 +1,121 @@
+// +build integ
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package multisigners
+
+import (
+	"testing"
+
+	csrctrl "istio.io/istio/pkg/test/csrctrl/controllers"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/tests/integration/security/util"
+)
+
+const (
+	ASvc = "a"
+	BSvc = "b"
+)
+
+type EchoDeployments struct {
+	Namespace namespace.Instance
+	// workloads for TestSecureNaming
+	A, B echo.Instances
+}
+
+var (
+	inst     istio.Instance
+	apps     = &EchoDeployments{}
+	stopChan = make(chan struct{})
+)
+
+func SetupApps(ctx resource.Context, apps *EchoDeployments) error {
+	var err error
+	apps.Namespace, err = namespace.New(ctx, namespace.Config{
+		Prefix: "test-ns",
+		Inject: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	builder := echoboot.NewBuilder(ctx)
+	builder.
+		WithClusters(ctx.Clusters()...).
+		WithConfig(util.EchoConfig(ASvc, apps.Namespace, false, nil)).
+		WithConfig(util.EchoConfig(BSvc, apps.Namespace, false, nil))
+
+	echos, err := builder.Build()
+	if err != nil {
+		return err
+	}
+	apps.A = echos.Match(echo.Service(ASvc))
+	apps.B = echos.Match(echo.Service(BSvc))
+	return nil
+}
+
+func TestMain(m *testing.M) {
+	framework.NewSuite(m).
+		Label(label.CustomSetup).
+		RequireSingleCluster().
+		Setup(istio.Setup(&inst, setupConfig)).
+		Setup(func(ctx resource.Context) error {
+			return SetupApps(ctx, apps)
+		}).
+		Run()
+	stopChan <- struct{}{}
```

Comment:
```suggestion
	close(stopChan)
```

otherwise if multiple things read it will be broken

Reply:
fixed